src/moq/messages.cpp

```diff
@@ -18,6 +18,42 @@ namespace moq::messages {
         return true;
     }
 
+    static bool ParseExtensions(qtransport::StreamBuffer<uint8_t>& buffer,
+                                uint64_t& count,
+                                std::optional<Extensions>& extensions,
+                                std::optional<uint64_t>& current_tag)
+    {
+        if (count == 0) {
+            return true;
+        }
+
+        if (extensions == std::nullopt) {
+            extensions = Extensions();
+        }
+
+        size_t completed = 0;
+        for (size_t extension = 0; extension < count; extension++) {
+            if (current_tag == std::nullopt) {
+                uint64_t tag{ 0 };
+                if (!ParseUintVField(buffer, tag)) {
+                    count -= completed;
+                    return false;
+                }
+                current_tag = tag;
+            }
+            auto val = buffer.DecodeBytes();
+            if (!val) {
+                count -= completed;
+                return false;
+            }
+            extensions.value()[current_tag.value()] = std::move(val.value());
+            current_tag = std::nullopt;
+            completed++;
+        }
+        count -= completed;
+        return true;
```

Comment:
Ok to merge, but note for later. It's unexpected that count (num_extensions) is set to 0 after successful parsing. Nothing uses num_extensions anymore after parsing completes, but it may be safer to set it back to the parsed number of extensions (or size of the map, or assert they are the same) in case the caller uses num_extensions later.